In [1]:
# "kelly.map" is a table extracted from the "execute_mut_accum_partially_recessive.slim"
# SLiM configuration file
recomb_map <- read.table("../kelly.map", col.names=c("pos", "rate"))

In [2]:
head(recomb_map)

pos        rate
1 7490641 0.000000010
2 7490643 0.000019060
3 7491792 0.000000010
4 7491794 0.000857735
5 7492660 0.000000010
6 7492662 0.000029310

### Issue #1
Why does the first region end at position 7490641? Based on the recombination map config format required by SLiM this would make the first exon 7490641 bp long.

### Issue #2

#### From the paper

> Specifically, for each pair of adjacent exons separated by a gap of b base pairs, we represent this gap as a single base pair with recombination rate b × 10−8 per generation. Similarly, each boundary between two chromosomes is encoded as a single base pair with a recombination rate of 0.5 crossovers per generation.

#### What is the minimum recombination rate for any region in the map?

Based on what's written in the paper, this shouldn't be less than 1e-8 (which corresponds to the recombination rate of a 1 bp "region" of the genome).

In [3]:
min(recomb_map$rate)

[1] 5e-09

How many of such sites are in the recombination map?

In [4]:
sum(recomb_map$rate < 1e-8)

[1] 55

#### What is the maximum recombination rate for any region in the map?

It shouldn't be higher than 0.5, which is the case for alleles that are completely unlinked (e.g. alleles on separate chromosomes that have a 50% chance of recombination).

Based on the information from the paper, there should be 21 positions with recombination rate of 0.5 (one for each pair of "adjacent" chromosomes).

In [5]:
sum(recomb_map$rate > 0.49)

[1] 0

There's no region with a recombination rate of 0.5, what's the maximum value then?

In [6]:
max(recomb_map$rate)

[1] 0.09321056